cartographer_ros/cartographer_ros/ros_map_writing_points_processor.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_ROS_ROS_MAP_WRITING_POINTS_PROCESSOR_H_
+#define CARTOGRAPHER_ROS_ROS_MAP_WRITING_POINTS_PROCESSOR_H_
+
+#include "cartographer/common/lua_parameter_dictionary.h"
+#include "cartographer/io/file_writer.h"
+#include "cartographer/io/points_processor.h"
+#include "cartographer/mapping_2d/proto/range_data_inserter_options.pb.h"
+#include "cartographer/mapping_2d/range_data_inserter.h"
+
+namespace cartographer_ros {
+
+// Very similar to Cartographer's ProbabilityGridPointsProcessor, but writes
+// out a PGM and YAML suitable for ROS map server to consume.
+class RosMapWritingPointsProcessor
+    : public ::cartographer::io::PointsProcessor {
+ public:
+  constexpr static const char* kConfigurationFileActionName = "write_ros_map";
+  RosMapWritingPointsProcessor(
+      double resolution,
+      const ::cartographer::mapping_2d::proto::RangeDataInserterOptions&
+          range_data_inserter_options,
+      ::cartographer::io::FileWriterFactory file_factory,
+      const string& filestem, PointsProcessor* next);
```

Comment:
PointsProcesser* const?

Reply:
Not in the header - it has no meaning for the API, it is purely an implementation detail.